### PR TITLE
Fix cert not working in ACR 

### PR DESCRIPTION
**Pull request by aman-harness**

### CODE REVIEW

File: cmd/kaniko-acr/main.go [hunk 1]
```diff
@@ -308,15 +309,27 @@ func getACRToken(tenantId, clientId, clientSecret, cert, registry string) (strin
...
+	}
+	err = os.Setenv(clientSecretKeyEnv, clientSecret)
+	if err != nil {
+		errors.Wrap(err, "failed to set env variable client secret")
+	}
+	err = os.Setenv(tenantKeyEnv, tenantId)
+	if err != nil {
+		errors.Wrap(err, "failed to set env variable tenant Id")
+	}
 	os.Setenv(certPathEnv, ACRCertPath)
```
Comment (shubham149) [inline cmd/kaniko-acr/main.go:324]: handle err here as well.

File: cmd/kaniko-acr/main.go [hunk 2]
```diff
@@ -308,15 +309,27 @@ func getACRToken(tenantId, clientId, clientSecret, cert, registry string) (strin
 	}
 
 	// TODO check for presence of file as well.
-	os.Setenv(clientIdEnv, clientId)
-	os.Setenv(clientSecretKeyEnv, clientSecret)
-	os.Setenv(tenantKeyEnv, tenantId)
+	err := os.Setenv(clientIdEnv, clientId)
```
Comment (shubham149) [inline cmd/kaniko-acr/main.go:312]: if err := os.Setenv(clientIdEnv, clientId); err != nil {
  return errors.Wrap(err, "failed to set env variable client Id")
}

Comment (shubham149) [reply cmd/kaniko-acr/main.go:312]: Same for all other cases

File: cmd/kaniko-acr/main.go [hunk 3]
```diff
@@ -368,7 +381,11 @@ func fetchACRToken(tenantId, token, registry string) (string, error) {
 }
 
 func setupACRCert(cert string) error {
-	err := ioutil.WriteFile(ACRCertPath, []byte(cert), 0644)
+	decoded, err := base64.StdEncoding.DecodeString(cert)
```
Comment (shubham149) [inline cmd/kaniko-acr/main.go:384]: Add a comment in the flag that it is expected that cert needs to be in base64 format as an input.

File: cmd/kaniko-acr/main.go [hunk 4]
```diff
@@ -368,7 +381,11 @@ func fetchACRToken(tenantId, token, registry string) (string, error) {
...
+	if err != nil {
+		return err
```
Comment (shubham149) [inline cmd/kaniko-acr/main.go:386]: return errors.Wrap(err, "failed to base64 decode ACR certificate")

File: cmd/kaniko-acr/main.go (hunk 1 above)
Comment (aman-harness) [reply cmd/kaniko-acr/main.go:324]: its handled already

File: cmd/kaniko-acr/main.go (hunk 3 above)
Comment (aman-harness) [reply cmd/kaniko-acr/main.go:384]: done

File: cmd/kaniko-acr/main.go (hunk 4 above)
Comment (aman-harness) [reply cmd/kaniko-acr/main.go]: done

File: cmd/kaniko-acr/main.go (hunk 2 above)
Comment (aman-harness) [reply cmd/kaniko-acr/main.go]: done

**Review by shubham149 — APPROVED** [review]

**Pull request merged**